### PR TITLE
Fix melody search window resize bugs

### PR DESCRIPTION
**Pull request by dchiller**
Addresses parts of #1128

### CODE REVIEW

**Comment by jacobdgm** [comment]
This is excellent - thank you for attending to this! 

(Looking forward to giving this a proper review)

File: django/cantusdb_project/main_app/templates/melody_search.html
Comment (jacobdgm) [inline django/cantusdb_project/main_app/templates/melody_search.html:16]: I can't find `text-truncate` anywhere else in the codebase. I gather this prevents too many characters (or `img`s(?)) from being added to the `div`, but is this bit of styling something that bootstrap provides?

File: django/cantusdb_project/static/js/melody_search.js
Comment (jacobdgm) [inline django/cantusdb_project/static/js/melody_search.js:46]: > Change drawArea coordinates on resize wo deleting current search

I'm curious about the "wo deleting current search" part of this commit message. Is there a gotcha here that could lead to the query or search results being lost if someone makes a minor tweak? If so, could a comment be added in the code explaining the tricky thing?

File: django/cantusdb_project/static/js/melody_search.js
Comment (jacobdgm) [inline django/cantusdb_project/static/js/melody_search.js:89]: Maybe we can leave this to a future PR, but it could be good to put this mapping in an object ("dictionary") - it would eliminate a bunch of `else if`s, and ensure that the values are always identical between `trackClick` and `trackMouse`.

**Review by jacobdgm — APPROVED** [review]
This all looks good - thanks for taking a look at and fixing this!

Approving because it's important to have it fixed as soon as we can. However, we know we'll need to wait until I've completed #1119 before any new changes make their way to staging or production, so I think there's time to address [this](https://github.com/DDMAL/CantusDB/pull/1130#pullrequestreview-1713249985) and [this](https://github.com/DDMAL/CantusDB/pull/1130#pullrequestreview-1713246177) before merging.

File: django/cantusdb_project/main_app/templates/melody_search.html
Comment (dchiller) [reply django/cantusdb_project/main_app/templates/melody_search.html:16]: Yes, this is bootstrap. To be honest, I don't quite understand why it worked, and I only saw this (https://getbootstrap.com/docs/5.0/helpers/text-truncation/) but didn't look at the underlying css. 

I think I will add a comment here...we can put it up for time's sake but we may want to clean it up later?

File: django/cantusdb_project/static/js/melody_search.js
Comment (dchiller) [reply django/cantusdb_project/static/js/melody_search.js:89]: Yes, good point! Maybe leave this to another PR. I noticed some other things that could be cleaned-up about this that I can add to make a more low priority issue.

**Comment by dchiller** [comment]
> However, we know we'll need to wait until I've completed #1119 before any new changes make their way to staging or production, so I think there's time to address [this](https://github.com/DDMAL/CantusDB/pull/1130#pullrequestreview-1713249985) and [this](https://github.com/DDMAL/CantusDB/pull/1130#pullrequestreview-1713246177) before merging.

Ah! Just saw this after commenting above...ok, I can go ahead and make those changes here then :) 

Did you pull this branch and test locally? I think it would be good for someone other than me to do that! (Or we can just wait for staging)

File: django/cantusdb_project/static/js/melody_search.js
Comment (dchiller) [reply django/cantusdb_project/static/js/melody_search.js:89]: Nvm...will make the change here.

File: django/cantusdb_project/static/js/melody_search.js
Comment (dchiller) [reply django/cantusdb_project/static/js/melody_search.js:46]: I will clarify...

**Comment by dchiller** [comment]
Haha! Too much going on at once...since this is now merged, I will put all the suggestions that I have and that @jacobdgm had in his review into another lower priority issue.

**Comment by jacobdgm** [comment]
> Ah! Just saw this after commenting above...ok, I can go ahead and make those changes here then :)

Oops, and your message appeared just after I pressed "merge".

> Did you pull this branch and test locally? I think it would be good for someone other than me to do that! (Or we can just wait for staging)

I haven't, but let me do this (now that it's on the develop branch lol). I'll report back once I've tested.

**Comment by jacobdgm** [comment]
in Safari and Chrome, it's working just as I'd expect. Excellent work - thank you!

**Pull request merged**
